firmware/temperature_detection/temperature_detection.ino

```diff
@@ -11,8 +11,8 @@ typedef enum {
 } indicator;
 
 typedef enum {
-    STATUS_FEVER_LOW = INDICATOR_GREEN,
-    STATUS_FEVER_HIGH = INDICATOR_RED,
+    STATUS_FEVER_LOW = INDICATOR_ORANGE | INDICATOR_GREEN,
+    STATUS_FEVER_HIGH = INDICATOR_ORANGE | INDICATOR_RED,
```

Comment:
Sorry, I guess this was never properly documented.

The state machine is tightly coupled to the indicator colors, and `INDICATOR_ORANGE` helps to differentiate both visually and programmatically the readout results from the distance indications.

```c
// Indicator values (binary flags)
INDICATOR_RED    = 0b001 // 0x1
INDICATOR_ORANGE = 0b010 // 0x2
INDICATOR_GREEN  = 0b100 // 0x4

// Statuses (combination of the indicator flags)
STATUS_FEVER_LOW      = 0b110 // ORANGE | GREEN
STATUS_FEVER_HIGH     = 0b011 // ORANGE | RED
STATUS_DISTANCE_RIGHT = 0b100 // GREEN
STATUS_DISTANCE_WRONG = 0b010 // ORANGE
STATUS_TRIGGER        = 0b100 // Same as DISTANCE_RIGHT but for manual (button) triggering
```

If you remove the `INDICATOR_ORANGE` flag from the fever readout statuses, `STATUS_FEVER_LOW` will be the same as `STATUS_DISTANCE_RIGHT`, and it could (though it currently doesn't) alter the control flow.

This indicator-centric status approach might be replaced by something saner if you feel it's necessary, but it was the what I could devise in the beginning, while we were only thinking about Arduino + LED devices.

***

*Note: now the display should mimic the traditional indicators with three vertical bars (one for each color): https://github.com/helpfulengineering/project-temperature-detection/blob/6696880c6690d10cb6cfca31cdacbeef2d0c4d38/firmware/temperature_detection/temperature_detection.ino#L153-L164.*

Reply:
> If you remove the INDICATOR_ORANGE flag from the fever readout statuses, STATUS_FEVER_LOW will be the same as STATUS_DISTANCE_RIGHT, and it could (though it currently doesn't) alter the control flow.
This indicator-centric status approach might be replaced by something saner if you feel it's necessary, but it was the what I could devise in the beginning, while we were only thinking about Arduino + LED devices.

Thanks for the details, Let me verify how your changes behave on the board.

Reply:
I have modified the code to implement your proposal and I have changed the screen size to fit with M5StickC. I will propose a separate PR for M5Stack.

We would need to discuss in a separate channel with the team about the orange led signification. I may think it would cause some misunderstandings on the results interpretation.

For me we can squash and merge.